eacl/target.go

```diff
@@ -54,10 +57,34 @@ func (t *Target) BinaryKeys() [][]byte {
 //
 // Each element of the keys parameter is a slice of bytes is a serialized compressed public key.
 // See [elliptic.MarshalCompressed].
+// Deprecated: use [Target.SetAccounts] instead.
 func (t *Target) SetBinaryKeys(keys [][]byte) {
 	t.keys = keys
 }
 
+// Accounts returns list of accounts to identify target subject in a binary format.
+//
+// Each element of the resulting slice is a [user.IDSize] byte account.
+// Use `user := user.ID(slice)` to decode it into a type-specific structure.
+//
+// Instead of an account there can also be a 33 byte public key binary.
+// See [keys.PublicKey].
+//
+// You should check the length for each element to determine correct type using.
+//
+// The value returned shares memory with the structure itself, so changing it can lead to data corruption.
+// Make a copy if you need to change it.
+func (t Target) Accounts() [][]byte {
+	return t.keys
+}
+
+// SetAccounts sets list of accounts to identify target subject.
+//
+// Each element of the accounts parameter is a slice of bytes of [user.ID].
+func (t *Target) SetAccounts(accounts [][]byte) {
```

Comment:
why not just `user.ID` then? it is even documented like this

Reply:
`SetBinaryKeys` is deprecated now and changing `SetAccounts` signature eliminates possibility to pass public keys

Reply:
do not see problems or do not understand. if we do not want users to use `SetBinaryKeys` why we want them to be able to pass keys via `SetAccounts` (a func that is even documented to be used with user.ID slices only)?

Reply:
i bet the reason is that we cannot keep getter and setter symmetry while having `Set([]user.ID)`. Field is `[][]byte` anyway, so it's pretty normal to accept/return value of this type. At the same time, we may provide functional helpers for those who desires to set `[]user.ID` (https://github.com/nspcc-dev/neofs-sdk-go/pull/592#discussion_r1696720643)

Reply:
> Field is [][]byte anyway

the field should be kept as long as we have such eALCs set in public networks. but new eACLs should be filled with `SetAccounts` and no keys should be accepted. at least that is how i see it

Reply:
> new eACLs should be filled with SetAccounts and no keys should be accepted

it's questionable which ones are _new_. For example, if i need to add one rule to my existing table w/o touching previously set records - i dont expect the system to deny such table. It worked with N records, i expect it to work with N+1 ones. Yeah, i _can_ refill they whole table, but this is inconvenient for me being a client

> the field should be kept as long as we have such eALCs set in public networks

weve been through this before - its impossible to track this in general. We need to clearly define the moment after which support will be cut. This should be either a system timestamp or next major protocol version

Reply:
why we deprecate if we gonna support it? to me, that should work this way: 0.43.0 release (node and ir) supports both ways, 0.44.0 supports only accounts. "supports" means accepting new tables, old ones should always work until we understand there are no such tables in the contract

anyway, what is your understanding of public keys deprecation? how long are we gonna support it? till 1.0.0 SDK release? what definition will it have? `SetAccounts(accounts [][]byte)` too?

Reply:
Keys will be supported for as long as needed.

Reply:
As for the method interface --- I'd really love to see more specific type here. It can be done symmetrically, the only problem is keys in existing eacls. Keys can be converted into standard accounts, but that can be misleading. We can treat them as separate entities though. We already have `Role` concept in the `Target` that is independent of `Keys`. So we can also have `Accounts` that are independent of them, you get keys (size-checked!) in `BinaryKeys()` and you get accounts from `Accounts()`, they can technically be mixed in the same eacl rule.

Reply:
all cases can be supported by functions around the most generic methods accepting/returning `[][]byte`, so currently proposed approach is the most preferable to me cuz it is the closest to the protocol